### PR TITLE
feat(dex.trades): add nomiswap binance smart chain

### PR DESCRIPTION
**Pull request by developerfred**
Brief comments on the purpose of your changes:

```

* Added NomiSwap dex trades on Bsc to dex.trades table.
* add  NMX token 
* Modified the dex.trades model
- Updated dbt_project.yml file
```

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [x] `coin_id` represents the ID of the coin on coinpaprika.com
* [x] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

File: models/nomiswap/nomiswap_bnb_schema.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        description: "Contract address of the token bought"
+        tests:
+          - dex_trades_token_bought:
+              dex_trades_seed: ref('dex_trades_seed')
+      - &token_sold_address
+        name: token_sold_address
+        description: "Contract address of the token sold"
+        tests:
+          - dex_trades_token_bought:
+              dex_trades_seed: ref('dex_trades_seed')
```
[the remark this answers is not in the document]
Comment (Hosuke) [reply models/nomiswap/nomiswap_bnb_schema.yml:75]: Also you can add seeds for other similar PR as well.

[the remark this answers is not in the document]
Comment (developerfred) [reply models/nomiswap/nomiswap_bnb_schema.yml:75]: I will do this now.

[the remark this answers is not in the document]
Comment (developerfred) [reply models/nomiswap/nomiswap_bnb_schema.yml:75]: @Hosuke [957c065](https://github.com/duneanalytics/spellbook/pull/2005/commits/957c06554fc40c60eddb5615a265629030b5d64d)

**Comment by Hosuke** [comment]
GM @developerfred , please verify if `nomiswap_bnb.Pair_evt_Swap` this table is available in dune v2 engine.

File: models/nomiswap/bnb/nomiswap_bnb_trades.sql [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    ,incremental_strategy = 'merge'
+    ,unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index', 'trace_address']
+    ,post_hook='{{ expose_spells(\'["bnb"]\',
+                                      "project",
+                                      "nomiswap",
+                                    \'["codingsh"]\') }}'
+    )
+}}
+
+{% set project_start_date = '2022-08-31' %}     -- select min(evt_block_time) from nomiswap_bnb.NomiswapPair_evt_Swap
```
Comment (Hosuke) [inline models/nomiswap/bnb/nomiswap_bnb_trades.sql:15]: I modify this date based on [the query result](https://dune.com/queries/1651513?d=1)

**Review by Hosuke — APPROVED** [review]
To whom it may concern, I have reviewed this PR and here is the summary:

- Metadata & naming conventions look good ✅
- Config blocks match uniqueness tests in schema files ✅
- Tests and seeds added for the new models ✅
- Incremental filters added when joining large tables ✅
- Contract addresses are lowercased ✅ 
- Inner join to large tables(bnb.transactions) as possible ✅

[Sample query results here](https://dune.com/queries/1651540?d=1)

File: seeds/dex/trades/dex_trades_seed.csv [hunk 3]
```diff
@@ -124,6 +124,10 @@ bnb,biswap,1,2022-11-22,0x9050a1e2f7aa392e39195b5b22382d595e94104b72699655b198c4
 bnb,biswap,1,2022-11-22,0xe2efd633720bd29322da7e39857e98b3c9904a1c55f9b3995441160983e06296,119,0xe9e7cea3dedca5984780bafc599bd69add087d56,43.898605549510836795,0x563ca064e41f3b5d80adeecfe49ab375fd7afbef,2756.0281
 bnb,biswap,1,2022-11-22,0x0b9ba3ee4a0b3f4db2baba70be527f5a50aea77423cd921106ec9f0ac8dcf169,115,0xbb4cdb9cbd36b01bd1cbaebf2de08d9173bc095c,0.018248512111984361,0xe9e7cea3dedca5984780bafc599bd69add087d56,4.837104443141354752 
 bnb,biswap,1,2022-11-22,0xaf03946d9b98e5d6e841bb50555208801e48d773a8dfdf1ecc203d943c27c603,350,0x2170ed0880ac9a755fd29b2688956bd959f933f8,6.247796965743473757,0x7130d2a12b9bcbfae4f2634d864a1ee1ce3ead9c,0.43849
+bnb,nomiswap,1,2022-11-22,0x2a23d2358cb76eb7be9b10e79e9ccd54ef267f7abd0410df1aa797aa15291a3f,139,0xd32d01a43c869edcd1117c640fbdcfcfd97d9d65,2233.297473430339233608,0x55d398326f99059ff775485246999027b3197955,318.237033328878863377
```
Comment (jeff-dude) [inline seeds/dex/trades/dex_trades_seed.csv:127]: @Hosuke @developerfred it appears none of these tx's are from nomiswap. please see [query](https://dune.com/queries/1690943?seed_t6c1ea=test_schema.git_cd0f07af_dex_trades_seed&model_t6c1ea=test_schema.git_cd0f07af_nomiswap_bnb_trades&blockchain_t6c1ea=bnb&project_t6c1ea=nomiswap&version_t6c1ea=1)

Comment (jeff-dude) [reply seeds/dex/trades/dex_trades_seed.csv:127]: the model doesn't match to the seed for any of the four rows, repeated across three different tests

Comment (developerfred) [reply seeds/dex/trades/dex_trades_seed.csv:127]: I will check here now, thanks to the report.

Comment (Hosuke) [reply seeds/dex/trades/dex_trades_seed.csv:127]: Moreover how do we find out the parameter `test_schema.git_cd0f07af_nomiswap_bnb_trades` for the query?
Currently I can only figure out cd0f07af at [$GIT_SHA macro definition in the dbt-test](https://github.com/duneanalytics/spellbook/actions/runs/3595335598/jobs/6054697367#step:8:5)

Comment (Hosuke) [reply seeds/dex/trades/dex_trades_seed.csv:127]: I see, so the model name `test_schema.git_cd0f07af_nomiswap_bnb_trades` can be found [here](https://github.com/duneanalytics/spellbook/actions/runs/3595335598/jobs/6054697367#step:11:41) and the seed name `test_schema.git_cd0f07af_dex_trades_seed` can be found [here](https://github.com/duneanalytics/spellbook/actions/runs/3595335598/jobs/6054697367#step:10:12). I am going to use this query to find out unmatched seeds in other PRs.

Comment (developerfred) [reply seeds/dex/trades/dex_trades_seed.csv:127]: @Hosuke @jeff-dude Is it necessary to change the router address? or can I make new queries?

Comment (developerfred) [reply seeds/dex/trades/dex_trades_seed.csv:127]: looks like nomiswap has a proxy router contract: https://bscscan.com/address/0x0df8c8652a1a634538f4593c34a219c0e4848440

I will work with the txids of this proxy

Comment (developerfred) [reply seeds/dex/trades/dex_trades_seed.csv:127]: NomiSwap Router: https://bscscan.com/address/0xd654953d746f0b114d1f85332dc43446ac79413d#code

Comment (developerfred) [reply seeds/dex/trades/dex_trades_seed.csv]: done

Comment (jeff-dude) [reply seeds/dex/trades/dex_trades_seed.csv]: > I see, so the model name `test_schema.git_cd0f07af_nomiswap_bnb_trades` can be found [here](https://github.com/duneanalytics/spellbook/actions/runs/3595335598/jobs/6054697367#step:11:41) and the seed name `test_schema.git_cd0f07af_dex_trades_seed` can be found [here](https://github.com/duneanalytics/spellbook/actions/runs/3595335598/jobs/6054697367#step:10:12). I am going to use this query to find out unmatched seeds in other PRs.

yes -- this is important. please use these gh action generated tables to run queries and test output. they persist for ~24 hours and should be taken advantage of, great developer tool for testing

**Comment by jeff-dude** [comment]
thanks both for getting this finalized

**Pull request merged**